crates/virtio-queue/src/lib.rs

```diff
@@ -707,13 +720,6 @@ impl<M: GuestAddressSpace> QueueState<M> {
         Ok(true)
     }
 
-    /// Goes back one position in the available descriptor chain offered by the driver.
-    /// Rust does not support bidirectional iterators. This is the only way to revert the effect
-    /// of an iterator increment on the queue.
-    pub fn go_to_previous_position(&mut self) {
```

Comment:
The `go_to_previous_position()` method assumes there's only one thread consuming descriptor chain from the available ring, so it would rewind. With the new `Queue` and `QueueState`, the assumption is not true anymore. That's why I have moved `go_to_previous_position()` to `AvailIter` because `AvailIter` is protected by a lock logically.

Reply:
Let's take an example:
1) thread A locks the queue, consume a descriptor head 1, advance `next` to 2, unlock
2) thread B locks the queue, consume a descriptor head 2, advance `next` to 3, unlock
3) thread A locks the queue, rewind the head to `1` by `go_to_previous_position()`, unlock.
Then descriptor  2 will be consumed twice.